maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/AbstractGeneratorMojo.java

```diff
@@ -294,6 +313,26 @@ static String getDefaultGoalPrefix( MavenProject project )
         return defaultGoalPrefix;
     }
 
+    /**
+     * Collects all dependencies having {@code org.apache.maven} group ID that are NOT in provided scope.
+     */
+    private Set<Artifact> mavenDependenciesNotInProvidedScope()
+    {
+        LinkedHashSet<Artifact> wrongScopedDependencies = new LinkedHashSet<>();
+
+        for ( Artifact dependency : project.getArtifacts() )
+        {
+            if ( "org.apache.maven".equals( dependency.getGroupId() )
```

Comment:
This leads to false positives: https://issues.apache.org/jira/browse/MPLUGIN-385?focusedCommentId=17460907&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17460907